qiskit/chemistry/drivers/pyscfd/integrals.py

```diff
@@ -104,36 +108,40 @@ def _check_molecule_format(val):
     return val
 
 
-def _calculate_integrals(mol, hf_method='rhf', conv_tol=1e-9, max_cycle=50, init_guess='minao'):
-    """Function to calculate the one and two electron terms. Perform a Hartree-Fock calculation in
-        the given basis.
-    Args:
-        mol (gto.Mole) : A PySCF gto.Mole object.
-        hf_method (str): rhf, uhf, rohf
-        conv_tol (float): Convergence tolerance
-        max_cycle (int): Max convergence cycles
-        init_guess (str): Initial guess for SCF
-    Returns:
-        QMolecule: QMolecule populated with driver integrals etc
-    Raises:
-        QiskitChemistryError: Invalid hf method type
-    """
-    enuke = gto.mole.energy_nuc(mol)
+def _create_mol_and_run_scf(atom, unit, charge, spin, basis, hf_method='rhf', conv_tol=1e-9,
+                            max_cycle=50, init_guess='minao', max_memory=None):
+    """Create molecule and run SCF calculation"""
 
-    if hf_method == 'rhf':
-        m_f = scf.RHF(mol)
-    elif hf_method == 'rohf':
-        m_f = scf.ROHF(mol)
-    elif hf_method == 'uhf':
-        m_f = scf.UHF(mol)
-    else:
-        raise QiskitChemistryError('Invalid hf_method type: {}'.format(hf_method))
+    atom = _check_molecule_format(atom)
+    hf_method = hf_method.lower()
+    if max_memory is None:
+        max_memory = param.MAX_MEMORY
+
+    try:
+        verbose = pylogger.QUIET
+        output = None
+        if logger.isEnabledFor(logging.DEBUG):
+            verbose = pylogger.INFO
+            file, output = tempfile.mkstemp(suffix='.log')
+            os.close(file)
+
+        mol = gto.Mole(atom=atom, unit=unit, basis=basis,
+                       max_memory=max_memory, verbose=verbose, output=output)
+        mol.symmetry = False
+        mol.charge = charge
+        mol.spin = spin
+        mol.build(parse_arg=False)
+        m_f, ehf, enuke = _scf_calculation(mol, hf_method=hf_method, conv_tol=conv_tol,
+                                           max_cycle=max_cycle, init_guess=init_guess)
+    except Exception as exc:
+        raise QiskitChemistryError('Failed electronic structure computation') from exc
+
+    return m_f, ehf, enuke, mol
+
+
+def _calculate_integrals_from_scf(m_f, ehf, enuke, mol, hf_method='rhf', coef_rot_matrix=None):
+    """Create QMolecule object containing the integrals from a run of SCF calculation"""
```

Comment:
This function is called from the driver, type hints and a docstring would be nice to have.